addons/udes_stock/models/stock_rule.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, fields, models, _
+
+
+class StockRule(models.Model):
+    _inherit = "stock.rule"
+
+    u_push_on_drop = fields.Boolean(
+        "Push on drop-off",
+        default=False,
+        help="When stock is dropped in the "
+        "from_location of this path, "
+        "automatically push it onwards.",
+    )
+
+    @api.model
+    def get_path_from_location(self, location):
+        """
+        Find a single stock.rule for which the given location is
+        a valid starting location.
```

Comment:
Can we add - if one not found return an empty record set

Reply:
> And also, if more are found, it is sorted by parent path? I'm not clear what this means :)

When I made the change I thought parent_path (Odoo 14) was similar to parent_left (which was used in Odoo 11), but they are quite different. In Odoo 11, the parent left value would be sorted in descending order. I guess as you said it doesn't make any sense to sort by parent path, maybe I should remove the sorted() line and just use the oldest stock.rule record?